src/nwb_conversion_tools/tools/hdmf.py

```diff
@@ -90,3 +91,66 @@ def _get_maxshape(self) -> tuple:
 
     def _get_data(self, selection: Tuple[slice]) -> np.ndarray:
         return self.data[selection]
+
+
+class ImagingExtractorDataChunkIterator(GenericDataChunkIterator):
+    """
+    Generic data chunk iterator for an ImagingExtractor object
+    primarily used when writing imaging data to an NWB file.
+    """
+
+    def __init__(
+        self,
+        imaging_extractor: ImagingExtractor,
+        buffer_gb: Optional[float] = None,
+        buffer_shape: Optional[tuple] = None,
+        chunk_mb: Optional[float] = None,
+        chunk_shape: Optional[tuple] = None,
+        display_progress: bool = False,
+        progress_bar_options: Optional[dict] = None,
+    ):
+        self.imaging_extractor = imaging_extractor
+
+        if buffer_gb is None and buffer_shape is None:
+            buffer_gb = 1.0
+
+        if buffer_shape is None:
+            buffer_shape = self._get_default_buffer_shape(buffer_gb=buffer_gb)
```

Comment:
This should be called after getting the default chunk shape (or otherwise manually set chunk shape): https://github.com/hdmf-dev/hdmf/blob/dev/src/hdmf/data_utils.py#L327